[FEATURE][MC] Token list with network logo and token name

tommasini opened the pull request:
**Description**
This pull request aims to improve the token list, showing the name of the tokens and the network logo, it removes the default '$0.00' of token balances and it's switching them to a skeleton loader.
It was developed unit tests to the components as well.

This PR also have performance improvement on how we fetch the tokens balances.



**Technical Details**

This PR for core repo aims to update the Tokens name:
https://github.com/MetaMask/core/pull/1127/files#diff-7ada7b099f6bcc6f3d4acd71f94b604636abd22cc556a76dbe7b3e610b1d9233

Draft created for performance improve
https://github.com/MetaMask/core/pull/1128


**Test Steps** (All of them are recorded in the Screenshots/Recording section)
- Detect tokens
- Switch account
- Switch network
- Kill and reopen the app
- Show 0 balance main token wallet
- Navigate to buy token
- Hide token (DAI Stable coin) on mainnet
- Import automatically the token (DAI Stable Coin) hidden by address
    - Address: 0x6B175474E89094C44Da98b954EedeAC495271d0F
- Import manually the token hidden with custom token import
    - Address: 0x6B175474E89094C44Da98b954EedeAC495271d0F
- Navigate to the asset page

Non-recorded but tested
* Add token after the swap
* add token with coin gecko API

**Screenshots/Recordings**
Full behaviour on light mode simulator iPhone 14 pro (iOS16.2)
* https://recordit.co/AntCh1rcaK
Full behaviour on dark mode physical device Huawei p30 pro (Android 11)
* https://recordit.co/H30qk9FcyH

**Test Cases**
User imports tokens via token autocomplete across supported networks: https://recordit.co/fRCLO6haNT
User imports tokens by contract address: https://recordit.co/kntnmF0N5n
User imports token on network A then switches to network B: https://recordit.co/nEedgiJEfp
User imports token via asset watcher: https://recordit.co/iXi6IXEwpH
User performs a swap via a dapp then imports token: http://recordit.co/iXtKpUAe2f 
(On the next both recording, there is only the part of the feature but I updated the app from a past version)
User updates the app to this feature with token detection off: https://recordit.co/SGyPxQ21Bl
User updates the app to this feature with token detection already enabled: https://recordit.co/G2iRCJQEEo
User imports an account to wallet then removes that imported account: https://recordit.co/iQEaWdhIPT

**To be improved**
* When the token list it's big, randomly times the endpoint fails (error 429) and we need to refresh the wallet view to update the token balances, we need a callback for this situation, any ideas are welcome

**Issue**

Progresses #

**Checklist**

* [ ] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

tommasini committed:
Token list updated

tommasini committed:
merge main

tommasini committed:
broke patch deleted

tommasini committed:
Tokens functional component and assets controllers patch working

tommasini committed:
Merge branch 'main' into feature/649-wallet-token-list-update

tommasini committed:
update tokens name when tokens are already imported

tommasini committed:
update tokens name when tokens are already imported

tommasini committed:
Revert "update tokens name when tokens are already imported"

This reverts commit 6157bf3c37d143228b35af1a8c58099d813afe9c.

tommasini committed:
patck to token detection controller

tommasini committed:
fix update token name if it was not defined already

tommasini committed:
add condition to update tokens name if the token is not on the token list

tommasini committed:
skeleton text and update patch

tommasini committed:
added unit tests and small refactor

tommasini committed:
remove useCallback

tommasini committed:
update assets controllers patch

tommasini committed:
merge main and solve conflicts

tommasini committed:
update behaviour for manual import tokens

sethkfman commented on app/components/UI/SearchTokenAutocomplete/index.tsx:
```diff
@@ -55,7 +55,7 @@ const SearchTokenAutocomplete = ({ navigation }: Props) => {
   const [selectedAsset, setSelectedAsset] = useState({});
   const [isSearchFocused, setIsSearchFocused] = useState(false);
-  const { address, symbol, decimals, image } = selectedAsset as any;
+  const { address, symbol, decimals, iconUrl, name } = selectedAsset as any;
```
Why was name added to the asset?

tommasini replied:
Thanks for the review!

When we select the asset from the auto complete asset list, we need the name of it to display on the user address token list state, for that reason we need the name, for when we press submit we `TokensController.addToken` with the name to our token list state

Let me know if this answers to the question 🙏🏼

sethkfman reviewed (approved):
LGTM - please answer and resolve the questions for clarity.

tommasini commented on app/components/Views/Wallet/index.tsx:
```diff
@@ -271,9 +272,10 @@ const Wallet = ({ navigation }: any) => {
       assets = [
         {
-          name: 'Ether', // FIXME: use 'Ether' for mainnet only, what should it be for custom networks?
+          name: getTicker(ticker) === 'ETH' ? 'Ethereum' : capitalize(ticker),
```
I followed the [design](https://www.figma.com/file/NI7m8KldyGDA1GanSY03fy/Latest-Mobile-IA%2FNAV-Redesign?node-id=6779-171168&t=17Qg9NJgqD96ysXa-0) blinded and it was 'Ethereum' on the token name, but you are right 'Ether' it's the token name! I will change it!

tommasini replied:
Changed !

tommasini committed:
change token name from ethereum to ether

tommasini committed:
added support for when tokens are added via wallet_watch asset api

tommasini committed:
update snapshot and small fix on the patch

tommasini committed:
fix bug when it imports the account on a fresh install andetect tokens

tommasini committed:
solve found issue when detected tokens

tommasini committed:
Merge branch 'main' into feature/649-wallet-token-list-update

witmicko committed:
switch to nativeevent (#5960)

tommasini committed:
Added translations for MC 0.5 (#5964)

Cal-L committed:
[FIX] Remove select address as a prop on App index.js (#5962)

* Remove select address as a prop on App index.js

* Update snapshot

dependabot[bot] committed:
Bump @xmldom/xmldom from 0.8.3 to 0.8.6 (#5959)

Bumps [@xmldom/xmldom](https://github.com/xmldom/xmldom) from 0.8.3 to 0.8.6.
- [Release notes](https://github.com/xmldom/xmldom/releases)
- [Changelog](https://github.com/xmldom/xmldom/blob/master/CHANGELOG.md)
- [Commits](https://github.com/xmldom/xmldom/compare/0.8.3...0.8.6)

---
updated-dependencies:
- dependency-name: "@xmldom/xmldom"
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump http-cache-semantics from 4.1.0 to 4.1.1 (#5669)

Bumps [http-cache-semantics](https://github.com/kornelski/http-cache-semantics) from 4.1.0 to 4.1.1.
- [Release notes](https://github.com/kornelski/http-cache-semantics/releases)
- [Commits](https://github.com/kornelski/http-cache-semantics/compare/v4.1.0...v4.1.1)

---
updated-dependencies:
- dependency-name: http-cache-semantics
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

NicolasMassart committed:
Extract test accounts to env vars (#5952)

* make detox import SRP from accounts.js file.
* extract account infos to env vars
* update e2e instructions and add vars in .js.env template

---------

Co-authored-by: Curtis <Curtis.David7@gmail.com>

NicolasMassart committed:
improv: add workflow triggers for tags on release and QA (#5902)

update notifications

sethkfman committed:
[UPDATE] pin sonar action to commit (#5967)

* pin the sonar action to a specific commit

* format sonar file

* remove extra line from file

sethkfman committed:
[UPDATE] sonar action configuration (#5972)

* pin the sonar action to a specific commit

* format sonar file

* remove extra line from file

* bump the sonar checkout action version which will use compliant version of node

* configure sonar analysis directors, test path, exclusions

* added project & file configurations

* remove test directory argument and added more exclusions

* added more exclusions

* added test inclusions and root directory for tests

Gudahtt committed:
Move `eth_sendTransaction` method handler to a separate module (#5968)

* Refactor `eth_sendTransaction` method handler

The method handler for `eth_sendTransaction` has been moved to a
separate module to simplify unit testing. Unit tests have been added
as well.

This module was written in TypeScript, which asked for some additional
validation of the method parameters. We now throw a more explicit
message when the params are missing, and when the transaction
parameters are not an object. Previously the method would still throw
an error in these scenarios, just with a less helpful message. The
change in error message should be the only functional change here.

This refactor is intended to make testing PR #5619 easier.

* Support `.cjs` file extensions

The Metro configuration has been updated to support `.cjs` file
extensions. The `@metamask/utils` dependency relies upon the library
`superstruct` which uses `.cjs` for its main exported module.

NicolasMassart committed:
fix: bundle video in app to prevent crash when not available (2nd attempt) (#5941)

- bundle video with embedded the subtitles in app
- add a readme in video folder to explain risks if video and subs files are moved or removed
- add a shell script running ffmpeg to regenerate the video
- remove useless constants and code related to remote SRP video resources.

segun committed:
Send to multisig address. (#5931)

* Send to multisig address.

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* Linter fix.

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* Revert wdio changes

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

---------

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

sethkfman commented:
@SamuelSalas can you run the e2e and confirm with QA Pass

sethkfman committed:
fixed iyarc format removed exclusions not needed, added exclusion for production value since the project is stilll maintained (#5978)

tommasini committed:
[IMPLEMENT] Use Terms Modal (#5242)

* user terms modal builded

* use terms modal logic implemented

* small fix on translations

* remove old terms and conditions

* remove ts ignore

* fix typo

* create a MandatoryModal component

* readme and update props to generic names

* update snapshot

* changing && to ternary operator for a more accurate ui rendering

* added margin and added component to the storybook

* implemented webview to show the ToU

* scale font size

* clean

* fix snapshot

* fix lint

* analytics

* modal mandatory more agnostic

* Clean up styles and storybook

* Clean up on accept logic

* logic on util file termsOfUse

* scrolling webview on Android and reusable modal not dismissable

* Implemented option scroll needed on ModalMandatory

* update source of use terms

* fix import

* remove margin bottom

* english translation updated

* fix button import

* modal mandatory not dismissable

* change icon name

* use terms modal on android for existing users

* ToU called only called on nav/app component

* Revert "ToU called only called on nav/app component"

This reverts commit 7f2e12fe164e085731495358fbb53d46d1e10f29.

* fix redirect on ToU modal to acessibility website

* fix track ecent

* E2e appium/5783 tou onboarding new app installs (#5951)

* Webdriverio and Detox test scripts for Term of use feature. Test scripts updated to have the term of use steps

* Small changes to fit browserstack

* Remove static-logos.js file changes

* move terms of use to before importing or create a new allet

* Update Detox and wdio test scripts to be compatible with the recept term of use changes

* Solve lint issues

---------

Co-authored-by: Cal Leung <cleun007@gmail.com>
Co-authored-by: SamuelSalas <samuel.salas.reyes@gmail.com>

SamuelSalas commented:
Detox test results:
Test Suites: 1 failed, 13 passed, 14 total
Tests:       3 failed, 136 passed, 139 total
Snapshots:   0 total
Time:        1702.919 s

e2e/specs/deeplinks.spec.js

tommasini committed:
main token name changed

SamuelSalas committed:
Fix failing wdio test scripts

SamuelSalas committed:
Solve merge conflicts

SamuelSalas commented:
Webdriver.IO Test pass

SamuelSalas committed:
Snapshots updated

tommasini commented:
DO NOT MERGE THIS PR

This PR will be closed in order for this [one](https://github.com/MetaMask/metamask-mobile/pull/6020)
